src/tasks/minions/minigames/titheFarmActivity.ts

```diff
@@ -0,0 +1,135 @@
+			fruit = 'logavano';
+		}
+
+		const harvestXp = determineHarvest * fruitXp;
+		const depositXp =
+			74 * 10 * fruitXp + (determineHarvest - 74) * 20 * fruitXp + 250 * fruitXp;
+		const farmingXp = harvestXp + depositXp;
+
+		const harvestStr = `${user} ${user.minionName} successfully harvested ${determineHarvest}x ${fruit} fruit and received ${farmingXp} Farming XP.`;
+		const completedStr = `You have completed the Tithe Farm ${titheFarmsCompleted +
+			1}x times. You now have ${titheFarmPoints + determinePoints} points to spend.`;
+
+		const userBank = user.settings.get(UserSettings.Bank);
+		let bonusXpMultiplier = 0;
+		let farmersPiecesCheck = 0;
+		if (bankHasItem(userBank, itemID(`Farmer's strawhat`), 1)) {
+			bonusXpMultiplier += 0.004;
+			farmersPiecesCheck += 1;
+		}
+		if (
+			bankHasItem(userBank, itemID(`Farmer's jacket`), 1) ||
+			bankHasItem(userBank, itemID(`Farmer's shirt`), 1)
+		) {
+			bonusXpMultiplier += 0.008;
+			farmersPiecesCheck += 1;
+		}
+		if (bankHasItem(userBank, itemID(`Farmer's boro trousers`), 1)) {
+			bonusXpMultiplier += 0.006;
+			farmersPiecesCheck += 1;
+		}
+		if (bankHasItem(userBank, itemID(`Farmer's boots`), 1)) {
+			bonusXpMultiplier += 0.002;
+			farmersPiecesCheck += 1;
+		}
+		if (farmersPiecesCheck === 4) bonusXpMultiplier += 0.005;
+
+		const bonusXp = farmingXp * bonusXpMultiplier;
+		const totalXp = farmingXp + bonusXp;
+
+		let bonusXpStr = '';
+		if (bonusXp > 0) {
+			bonusXpStr = `You received an additional ${Math.floor(
+				bonusXp
+			)} Bonus XP from your farmer's outfit pieces.`;
+		}
+
+		await user.addXP(SkillsEnum.Farming, Math.floor(totalXp));
+
+		if (roll((7494389 - user.skillLevel(SkillsEnum.Farming) * 25) / determineHarvest)) {
+			const loot = { [itemID('Tangleroot')]: 1 };
+			lootStr += '\n\n```diff';
+			lootStr += `\n- You have a funny feeling you're being followed...`;
+			lootStr += '```';
+			this.client.emit(
+				Events.ServerNotification,
+				`${Emoji.Farming} **${user.username}'s** minion, ${
+					user.minionName
+				}, just received a Tangleroot by completing the Tithe Farm on their ${titheFarmsCompleted +
+					1} run!`
+			);
+
+			await user.addItemsToBank(loot, true);
+		}
+
+		if (!channelIsSendable(channel)) return;
+		const returnStr = `${harvestStr} ${bonusXpStr}\n\n${completedStr}${lootStr}\n\n${user.minionName} asks if you'd like them to do another game.`;
+
+		msg.author.incrementMinionDailyDuration(duration);
+
+		this.client.queuePromise(() => {
+			channel.send(returnStr);
+
+			channel
+				.awaitMessages(mes => mes.author === user && saidYes(mes.content), {
+					time: getUsersPerkTier(user) > 1 ? Time.Minute * 10 : Time.Minute * 2,
+					max: 1
+				})
+				.then(messages => {
+					const response = messages.first();
+
+					if (response) {
+						if (response.author.minionIsBusy) return;
+
+						user.log(`attempted another run of Tithe Farm`);
+
+						this.client.commands.get('tithefarm')!.run(response as KlasaMessage, []);
+					}
```

Comment:
this whole section isnt needed anymore due to recent refactoring - checkout a task to see the difference.

**however** dont remove the `msg.author.incrementMinionDailyDuration(duration);` line, , move it to the top and use `user` instead of `msg.author`